src/components/modebar/manage.js

```diff
@@ -99,10 +99,19 @@ function getButtonGroups(fullLayout, buttonsToRemove, buttonsToAdd) {
 
     var hasCartesian = fullLayout._hasCartesian,
         hasGL2D = fullLayout._hasGL2D,
-        allAxesFixed = areAllAxesFixed(fullLayout);
+        allAxesFixed = areAllAxesFixed(fullLayout),
+        dragModeGroup = [];
+
+    if((hasCartesian || hasGL2D) && !allAxesFixed) {
+        dragModeGroup = ['zoom2d', 'pan2d'];
+    }
+    if(hasCartesian) {
+        dragModeGroup.push('select2d');
+        dragModeGroup.push('lasso2d');
```

Comment:
Selection does have a use even without events or linking, for highlighting data while discussing or presenting it. And just because it's cool to play with :sunglasses: So I wouldn't leave it entirely to developers to enable.

Then re: diving into fullData, I guess it depends how fast you think we'll add other trace types. (not sure if I'll be able to do it myself, but now it's just a matter of adding `selectPoints` methods to other modules) It seems pretty clear what select means for heatmaps and contour maps, and (at least for 1D as @chriddyp suggests, and I'm sure we can agree on 2D as well) what it means for bars, histograms, and boxes.